addon/caluma-query/queries/form.js

```diff
@@ -0,0 +1,91 @@
+import gql from "graphql-tag";
+
+import BaseQuery from "./base";
+
+export default class FormQuery extends BaseQuery {
+  dataKey = "allForms";
+  modelName = "form";
+
+  get query() {
+    // query next page of forms that satisfy filters
+    return `
+      query (
+        $filter: [FormFilterSetType], 
+        $order: [FormOrderSetType], 
+        $pageSize: Int, 
+        $cursor: String
+      ) {
+        allForms(
+          filter: $filter, 
+          order: $order, 
+          first: $pageSize,
+          after: $cursor,
+        ) {
+          ${this.pagination}
+          edges {
+            node {
+              id
+              __typename
+              ...FormInfo
+            }
+          }
+        }
+      }
+
+      fragment FormInfo on Form ${this.modelClass.FormInfo} 
+
+    `;
+  }
+
+  get searchQuery() {
+    // query all form slugs
+    return `
+        query {
+          allForms {
+            edges {
+              node {
+                slug
+              }
+            }
+          }
+        }
+      `;
+  }
+
+  fetch({ filter = [], order = [], search = "", queryOptions = {} } = {}) {
+    if (search) {
+      // pre-filter forms based on slugs containing search value
+      // before performing pagination
+      return this.search({ filter, order, search, queryOptions });
+    }
+
+    // perform form fetch with pagination
+    return super.fetch({ filter, order, queryOptions });
+  }
+
+  async search({
+    filter = [],
+    order = [],
+    search = "",
+    queryOptions = {},
+  } = {}) {
+    // query all form slugs
+    let forms = await this.apollo.query(
+      {
+        query: gql`
+          ${this.searchQuery}
+        `,
+        fetchPolicy: "network-only",
+      },
+      "allForms.edges"
+    );
+
+    // select forms whose slug contains search value and
+    // use slugs as filter for pagination query
+    forms = forms.filter((form) => form.node.slug.includes(search));
+    filter.push({ slugs: forms.map((form) => form.node.slug) });
+
+    // perform form fetch with pagination
+    return super.fetch({ filter, order, queryOptions });
+  }
```

Comment:
IMO this should not be necessary. `search` can be used like every other filter provided by the API. You should be able to just call `fetch({ filter: [{ search: "xy" }] })`